Common/src/Pollster/TaskHandler.cs

```diff
@@ -641,10 +647,19 @@ await agent_.FinalizeTaskCreation(CancellationToken.None)
       }
       else
       {
-        logger_.LogError(e,
-                         resubmit
-                           ? "Error during task execution, retrying task"
-                           : "Error during task execution, cancelling task");
+        if (resubmit)
+        {
+          logger_.LogError(e,
+                           "Error during task execution, retrying task");
+          messageHandler_.Status = QueueMessageStatus.Cancelled;
+        }
+        else
+        {
+          logger_.LogError(e,
+                           "Error during task execution, cancelling task");
+          messageHandler_.Status = QueueMessageStatus.Processed;
```

Comment:
This changed the semantic a bit: if `CompleteTaskAsync`, then the message would still be acknowledged.